### PR TITLE
Pythonise "Working with Python arrays" (array.rst)

### PR DESCRIPTION
**Pull request by 0dminnimda**
Continuing to Pythonize Сython documentation!

This PR Pythonises [this](https://cython.readthedocs.io/en/latest/src/tutorial/array.html) documentation page.

Lots of process details and standardization are found here: #4187

### CODE REVIEW

**Comment by 0dminnimda** [comment]
@scoder, could you please review this?

File: docs/src/tutorial/array.rst [hunk 1]
```diff
@@ -55,13 +84,30 @@ it is possible to create a new array with the same type as a template,
...
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/array/resize.py
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/array/resize.pyx
 
 
 API reference
```
[the remark this answers is not in the document]
Comment (0dminnimda) [reply docs/src/tutorial/array.rst:113]: So, @scoder [it seems](https://github.com/cython/cython/pull/4508#discussion_r773453189) you would like to have pythonised version of the reference?

**Comment by matusvalo** [comment]
@0dminnimda are you still willing to finish this PR? I think the only pending point is to finish the Reference part. If yes I will close my PR in favor of yours.

**Comment by 0dminnimda** [comment]
> are you still willing to finish this PR?

Sure

File: docs/src/tutorial/array.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (0dminnimda) [reply docs/src/tutorial/array.rst:113]: done  74cf48016e46ee172813532fb9a9a214b54d7594

**Comment by 0dminnimda** [comment]
> I think the only pending point is to finish the Reference part.

ok, we're done with that

I wouldn't say I like this change, but for the sake of improving the docs, I won't argue about it.

image
image
image

**Comment by 0dminnimda** [comment]
Now the pr is ready for review, waiting for @scoder

**Comment by scoder** [comment]
Very nice. Thanks!

**Pull request merged**